gateway-service/src/main/java/org/zowe/apiml/gateway/loadbalancer/DeterministicLoadBalancer.java

```diff
@@ -114,12 +115,32 @@ private boolean isTooOld(LocalDateTime cachedDate) {
 
     private Mono<String> getSub(Object requestContext) {
         if (requestContext instanceof RequestDataContext ctx) {
-            var token = Optional.ofNullable(ctx.getClientRequest().getCookies().get("apimlAuthenticationToken")).map(list -> list.get(0)).orElse("");
+            var token = Optional.ofNullable(getTokenFromCookie(ctx))
+                                .orElseGet(() -> getTokenFromHeader(ctx));
             return Mono.just(extractSubFromToken(token));
         }
         return Mono.just("");
     }
 
+    private String getTokenFromCookie(RequestDataContext ctx) {
+        var tokens = ctx.getClientRequest().getCookies().get("apimlAuthenticationToken");
```

Comment:
You might consider using  Optional.ofNullable here and also in the `getTokenFromHeader` method. 
```
private String getTokenFromCookie(RequestDataContext ctx) {
    return Optional.ofNullable(ctx.getClientRequest().getCookies().get("apimlAuthenticationToken"))
                   .filter(tokens -> !tokens.isEmpty())
                   .map(tokens -> tokens.get(0))
                   .orElse(null);
}
```

Reply:
yes I just didn't want to nest the optionals, I kept the optional one level above

Reply:
Oh right, I missed it was above, ok looks good